packages/components/tests/dummy/app/controllers/components/table.js

```diff
@@ -148,4 +156,49 @@ export default class ComponentsTableController extends Controller {
     }
     return clonedModelClusters;
   };
+
+  // TODO refactor/rename/reorganize the code later
+
+  @action
+  onFilterChange(event) {
+    // console.log(event.target.value);
+    this.customFilterRows = event.target.value;
+  }
+
+  get filteredSelectableData() {
+    if (this.customFilterRows === 'all') {
+      return this.model.selectableData;
+    } else {
+      const reminder = this.customFilterRows === 'even' ? 0 : 1;
+      return this.model.selectableData.filter(
+        (item) => item.id % 2 === reminder
+      );
+    }
+  }
+
+  get paginatedSelectableDataTotalItems() {
+    return this.model.selectableData.length;
+  }
+
+  get paginatedSelectableData() {
+    const start =
+      (this.currentPaginatedSelectablePage - 1) *
+      this.currentPaginatedSelectablePageSize;
+    const end =
+      this.currentPaginatedSelectablePage *
+      this.currentPaginatedSelectablePageSize;
+    return this.model.selectableData.slice(start, end);
+  }
+
+  @action
+  onPaginatedSelectablePageChange(page) {
+    this.currentPaginatedSelectablePage = page;
+  }
+
+  @action
+  onPaginatedSelectablePageSizeChange(pageSize) {
+    // we agreed to reset the pagination to the first element (any alternative would result in an unpredictable UX)
```

Comment:
@KristinLBradley Does this mean, like the example in the showcase, that selecting the parent checkbox on page 1 will select all checkboxes, but on page 2, the parent checkbox will reset to unchecked, and so will the child ones?

Reply:
@andgen404 This is the current behaviour, and could be the "safest" one. I'll give you an example. Let's say I am an admin and I on a page with a list of users. I see a table, with a pagination. There are 40 records (users), I see 10 of them (4 pages) so I see 10 rows in the table. Now, I click "select all" and all the rows are selected. If I click the button "delete" next to the table of users, what should I expect to happen? That I delete the 10 users that I see (and only them) or that I delete all the 40 users (even if I don't see them?) 
Let's imagine I intended to only delete the ones that I see on screen, how would I know that the application instead is deleting all of them (because "secretly" has selected also the rows on the other pages)?
Now, if this is the case, probably we want to avoid this dangerous case, and when "select all" is clicked, only the visible rows are selected, right? But then what happens if the user clicks "select all" on the first page (so the "select all" becomes checked, and all the rows are checked) and then clicks on the next page? Well, according to the "safe" logic above, the only option we have is to show the rows in the second table's "page" as unselected, as well as the "select all". Do you see any alternatives to this?

Reply:
When I've used similar features features such as for managing email, it only selects the items visible on the page as Cristiano. Personally, I think that's the safest behavior and seems to be common.

Reply:
Ah, okay. Thank you, @didoo, for the example! I don't see any alternatives; I agree with this "Safe" approach as well.

Reply:
> similar features features such as for managing email

Gmail inbox behaves exactly like this :)

Reply:
I think for "select all" it should mean "select all visible rows" (and possibly even give a row count - "select all 20 rows")

Then there's an upgrade to that, an additional option to select all rows, even non-visible ones ("select all 100 rows in this table")

Reply:
As discussed, for this release we will settle on `Select all rows` / `Deselect all rows`.